src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -846,18 +849,37 @@ public void Close()
                 _machineName = ".";
                 _raisedOnExited = false;
 
-                //Don't call close on the Readers and writers
-                //since they might be referenced by somebody else while the 
-                //process is still alive but this method called.
-                _standardOutput = null;
-                _standardInput = null;
-                _standardError = null;
+                // Only call close on the streams if the user cannot have a reference on them.
+                // If they are referenced it is the user's responsibility to dispose of them.
+                try 
+                {
+                    if (_standardOutput != null && (_outputStreamReadMode == StreamReadMode.AsyncMode || _outputStreamReadMode == StreamReadMode.Undefined))
```

Comment:
In Mono, since the AsyncMode streams are not accessible, it also does CancelOutputRead() and Close() on those. Although it is not clear what this gets since we are about to close the underlying stream it is a bit cleaner and seems like we should do that.
Also please add a similar comment to Mono explaining why it is safe to close in AsyncMode (ie. ,because we don't expose the async streams)

Reply:
No Close() on the asyncstreamreader in .NET Core, but definitely will cancel the current operation.

Reply:
Done :)